assembl/tests/views_tests/test_graphql_vote_session.py

```diff
@@ -0,0 +1,254 @@
+            self.filename = filename
+            self.type = type
+
+    new_image_name = u'new-image.png'
+    new_image_mime_type = 'image/png'
+    new_image = FieldStorage(new_image_name, new_image_mime_type)
+    image_var_name = u'variables.img'
+    graphql_request.POST[image_var_name] = new_image
+
+    response = schema.execute(
+        fragments + u"""
+            mutation UpdateVoteSession(
+              $discussionPhaseId: Int!
+              $headerImage: String
+              $titleEntries: [LangStringEntryInput]
+              $subTitleEntries: [LangStringEntryInput]
+              $instructionsSectionTitleEntries: [LangStringEntryInput]
+              $instructionsSectionContentEntries: [LangStringEntryInput]
+              $propositionsSectionTitleEntries: [LangStringEntryInput]
+            ) {
+              updateVoteSession(
+                discussionPhaseId: $discussionPhaseId
+                headerImage: $headerImage
+                titleEntries: $titleEntries
+                subTitleEntries: $subTitleEntries
+                instructionsSectionTitleEntries: $instructionsSectionTitleEntries
+                instructionsSectionContentEntries: $instructionsSectionContentEntries
+                propositionsSectionTitleEntries: $propositionsSectionTitleEntries
+              ) {
+                voteSession {
+                  ...voteSessionGlobals
+                  ...voteSessionLangstringsEntries
+                }
+              }
+            }
+        """,
+        context_value=graphql_request,
+        variable_values={
+            "discussionPhaseId": discussion_phase_id,
+            "titleEntries": en_entry(new_title),
+            "subTitleEntries": en_entry(new_sub_title),
+            "instructionsSectionTitleEntries": en_entry(new_instructions_section_title),
+            "instructionsSectionContentEntries": en_entry(new_instructions_section_content),
+            "propositionsSectionTitleEntries": en_entry(new_propositions_section_title),
+            "headerImage": image_var_name
+        }
+    )
+
+    assert response.errors is None
+    fetched_vote_session = response.data['updateVoteSession']['voteSession']
+
+    assert en_value(fetched_vote_session['titleEntries']) == new_title
+    assert en_value(fetched_vote_session['subTitleEntries']) == new_sub_title
+    assert en_value(fetched_vote_session['instructionsSectionTitleEntries']) == new_instructions_section_title
+    assert en_value(fetched_vote_session['instructionsSectionContentEntries']) == new_instructions_section_content
+    assert en_value(fetched_vote_session['propositionsSectionTitleEntries']) == new_propositions_section_title
+
+    fetched_image = fetched_vote_session['headerImage']
+    assert fetched_image['title'] == new_image_name
+    assert fetched_image['mimeType'] == new_image_mime_type
+    new_image_data = new_image.file.getvalue()
+    fetched_image_data = test_app.get(fetched_image['externalUrl']).body
```

Comment:
Ah cool, so you're committing a LocalStorage object in here. So, as it stands, this object will NOT be deleted from the db when the test is complete. This breaks test isolation principles. Let's make a cleanup function to remove these excess objects from the DB after the test (like the attachment)